python/lsst/ts/m2gui/model.py

```diff
@@ -39,12 +40,16 @@ class Model(object):
         A logger.
     is_csc_commander : `bool`
         Commandable SAL component (CSC) is the commander or not.
-    local_mode : `LocalMode`
+    local_mode : enum `LocalMode`
         Local model.
     is_closed_loop : `bool`
         Closed-loop control is on or not.
-    signal_error : `SignalError`
-        Signal to report the new or cleared errors.
+    signal_status : `SignalStatus`
+        Signal to report the updated status of system.
+    system_status : `dict`
+        System status.
+    fault_manager : `FaultManager`
+        Fault manager to record the system error.
```

Comment:
The problem I see there is that you aren't developing a web application, where MVC paradigm would make a sense. Much better is to combine model with signals. Instead of listing signal_status etc. as attribute, why not create it directly in Model?
```python

class Model(QObject):
    ....
    signal_status = QtCore.Signal(str, bool)
    ....

    def __init__(self,..):
         .... 
```

That will remove extra files in the signal, which are just annoyances making code harder to read. Of course, you will be able to use Model.system_status to hook for updates (connect).

Reply:
Please see  https://www.reddit.com/r/QtFramework/comments/fmlw2o/has_anyone_created_a_proper_mvc_pattern_using_qt/ for some discussion.

Reply:
I think you are right for the multiple signal files. I created a new `signals.py` file to hold all the signals and remove the extra files in commit: 9b0e18c. Thanks to point out this!

At this moment, I do not really decide which class should hold which signal yet. Therefore, I prefer to have the classes of signals because it is easier to modify the code and related design.

I prefer to use the `Model` class to handle the business logic. One thing I do not like the LabVIEW is to let the `View` to handle the business logic. Therefore, I prefer to have some decoupling at this moment.

Thanks!